README.md

```diff
@@ -420,6 +421,93 @@ status:
+```
+
+As soon as you run this, you'll start seeing some output being streamed to STDOUT.
+
+#### Phase 1: Initializing deploy
+
+In this phase, we:
+
+- Perform basic validation to ensure we can proceed with the deploy. This includes checking if we can reach the context, if the context is valid, if the namespace exists within the context, and more. We try to validate as much as we can before trying to ship something because we want to avoid having an incomplete deploy in case of a failure (this is especially important because there's no rollback support).
+- List out all the resources we want to deploy (as described in the template files we used).
+- Render ERB templates and apply partials, if enabled (which is the case for this example). If enabled, we also perform basic validation on the parsed templates.
+
+#### Phase 2: Checking initial resource statuses
+
+In this phase, we check resource statuses. For each resource listed in the previous step, we check Kubernetes for their status; in the first deploy this might show a bunch of items as "Not Found", but for the deploy of a new version, this is an example of what it could look like:
+
+```
+Certificate/services-foo-tls     Exists
+Cloudsql/foo-production          Provisioned
+Deployment/jobs                  3 replicas, 3 updatedReplicas, 3 availableReplicas
+Deployment/web                   3 replicas, 3 updatedReplicas, 3 availableReplicas
+Ingress/web                      Created
+Memcached/foo-production         Healthy
+Pod/db-migrate-856359            Unknown
+Pod/upload-assets-856359         Unknown
+Redis/foo-production             Healthy
+Service/web                      Selects at least 1 pod
+```
+
+The next phase might be either "Predeploying priority resources" (if there's any) or "Deploying all resources". In this example we'll go through the former, as we do have predeployable resources.
+
+#### Phase 3: Predeploying priority resources
+
+In this phase we predeploy certain types of resources (e.g. `ConfigMap`, `PersistentVolumeClaim`, `Secret`) to make sure the latest version will be available when resources that might consume them (e.g. `Deployment`) are deployed. This will not show up if the templates don't include resources of a type that would be relevant here.
+
+When this runs, we essentially run `kubectl apply` on those templates and periodically check the cluster for the current status of each resource so we can display error or success information. This will look different depending on the type of resource. If you're running the command described above, you should see something like this in the output:
+
+```
+Deploying ConfigMap/hello-cloud-configmap-data (timeout: 30s)
+Successfully deployed in 0.2s: ConfigMap/hello-cloud-configmap-data
+
+Deploying PersistentVolumeClaim/hello-cloud-redis (timeout: 300s)
+Successfully deployed in 3.3s: PersistentVolumeClaim/hello-cloud-redis
+
+Deploying Role/role (timeout: 300s)
```

Comment:
300 is strange here. 

```
module KubernetesDeploy
  class Role < KubernetesResource
    TIMEOUT = 30.seconds
```

Reply:
It's reproducible though. I'll try to debug this and, if needed, will fix on a separate PR.